File: web/tests/reference/sliding-window/minimum-size-subarray-sum.js
function minSubArrayLen(target, nums) {
  let left = 0;
  let windowSum = 0;
  let best = Infinity;
  for (let right = 0; right < nums.length; right++) {
    windowSum += nums[right];
    while (windowSum >= target) {
      best = Math.min(best, right - left + 1);
      windowSum -= nums[left];
      left++;
    }
  }
  return best === Infinity ? 0 : best;
}

File: web/tests/reference/sliding-window/minimum-size-subarray-sum.py
def min_subarray_len(target, nums):
    left = 0
    window_sum = 0
    best = float("inf")
    for right in range(len(nums)):
        window_sum += nums[right]
        while window_sum >= target:
            best = min(best, right - left + 1)
            window_sum -= nums[left]
            left += 1
    return 0 if best == float("inf") else best
